ruby-challenges.rb

```diff
@@ -10,23 +10,126 @@
 # HINT: Google 'ruby get rid of nested arrays'
 
-us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Vermont'] }
+us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], northeast: ['Maine', 'New Hampshire', 'Vermont'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Vermont', 'Washington'] 
 
+def state_selector(hash)
+    hash.values.flatten.sort
+end
+
+p state_selector(us_states)
+
+# Method: 
+# def state_selector(hash)
+#     hash
+#         --> returns hash
+#     hash.values
+#         --> removes nested arrays
+#     hash.values.flatten
+#         --> returns all hash values as one array
+#     hash.values.flatten.sort
+#         --> sorts the values alphabetically
+# end
+
+
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
-# Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
+class Bike
+    attr_accessor :model, :wheels, :current_speed
+    def initialize(model, wheels, current_speed)
+        # attr_accessor to set instance variables: model, wheels, and current_speed
+        @model = model
+        @wheels = 2
+        @current_speed = 0
+    end
+
+    def bike_info
+        "I'm riding a #{@model} bike with #{@wheels} wheels, going #{@current_speed} mph."
+        # bike_info method using string interpolation to call @model, @wheels, and @current speed
+    end
+end
 
+bmx_bike = Bike.new('BMX', @wheels, @current_speed)
+    # new object bmx_bike just need to set a new string for the model, as wheels and current_speed are already set
 
+p bmx_bike.bike_info
+# output: "I'm riding a BMX bike with 2 wheels, going 0 mph."
+    # prints correctly
 
 # -------------------3b) Add the ability to pedal faster and brake. The pedal_faster method should increase the speed by a given amount. The brake method should decrease the speed by a given amount. The bike cannot go negative speeds.
 
 # Expected output example: my_bike.pedal_faster(10) => 10
 # Expected output example: my_bike.pedal_faster(18) => 28
 # Expected output example: my_bike.brake(5) => 23
 # Expected output example: my_bike.brake(25) => 0
+
+# * Resouces used *
+# https://www.geeksforgeeks.org/ruby-numeric-negative-function/
+    # info on how to use negative?() built in method in order to prevent the brake method from going into negative numbers 
+
+class Tricycle
+    attr_accessor :model, :wheels, :current_speed
+    def initialize(model, wheels, current_speed)
+        @model = model
+        @wheels = wheels
```

Comment:
Since the prompt mentioned wheels would default to 2 wheels, you don't need to pass a param to this, you can just set the instance variable to 2.